intel-sgx/dcap-artifact-retrieval/src/cli.rs

```diff
@@ -0,0 +1,223 @@
+            (
+                @arg ORIGIN: --("origin") +takes_value
+                validator(|s| parse_origin(s.as_str()).map(|_| ()))
+                "Location from where artifacts need to be fetched. Options are: \"intel\" and \"azure\".\
+                 Note that Azure does not provide access to all artifacts. Intel will be contacted as a fallback (default: \"intel\")"
+            )
+            (
+                @arg PCKID_FILE: --("pckid-file") +takes_value +required requires("PCKID_FILE")
+                validator(is_pckid_file)
+                "File describing the PCK identity (outputted by PCKIDRetrievalTool)"
+            )
+            (
+                @arg OUTPUT_DIR: --("output-dir") +takes_value +required requires("OUTPUT_DIR")
+                validator(is_directory)
+                "Destination folder for data retrieved from Intel certification services"
+            )
+            (
+                @arg API_VERSION: --("api-version") +takes_value
+                validator(|s| parse_pcs_version(s.as_str()).map(|_| ()))
+                "API version for provisioning service, supported values are 3 and 4 (default: 3)"
+            )
+            (
+                @arg API_KEY: --("api-key") +takes_value
+                "API key for authenticating with Intel provisioning service"
+            )
+            (
+                @arg VERBOSE: -v --verbose
+                "Print information of which files are fetched"
+            )
+        )
+        .get_matches();
+
+    let result = match (
+        matches.value_of("PCKID_FILE"),
+        matches.value_of("OUTPUT_DIR"),
+    ) {
+        (Some(pckid_file), Some(output_dir)) => {
+            let verboseness = matches.occurrences_of("VERBOSE");
+            let api_version = parse_pcs_version(matches.value_of("API_VERSION").unwrap_or("3"))
+                .expect("validated");
+
+            let origin =
+                parse_origin(matches.value_of("ORIGIN").unwrap_or("intel")).expect("validated");
+
+            let fetcher = crate::reqwest_client();
+            let client: Box<dyn ProvisioningClient> = match origin {
+                Origin::Intel => {
+                    let mut client_builder = IntelProvisioningClientBuilder::new(api_version);
+                    if let Some(api_key) = matches.value_of("API_KEY") {
+                        client_builder.set_api_key(api_key.into());
+                    }
+                    Box::new(client_builder.build(fetcher))
+                }
+                Origin::Azure => {
+                    let client_builder = AzureProvisioningClientBuilder::new(api_version);
+                    Box::new(client_builder.build(fetcher))
+                }
+            };
+            download_dcap_artifacts(&*client, pckid_file, output_dir, 0 < verboseness)
+        }
+        _ => unreachable!("validated"),
+    };
+
+    match result {
+        Ok(()) => {}
+        Err(Error::PCSError(StatusCode::NotFound, _)) => {
```

Comment:
Nit: I suggest to also print error.

Reply:
> Nit: I suggest to also print error.

I don't think there is more information in the message parameter after checking the code.